lib/bloc/location/location_bloc.dart

```diff
@@ -15,21 +17,104 @@ class LocationBloc extends Bloc<LocationEvent, LocationState> {
         (LocationEvent event, Emitter<LocationState> emit) async {
       await _requestCurrentLocation(emit);
     });
+    on<RequestPermissionLocationEvent>(
+        (LocationEvent event, Emitter<LocationState> emit) async {
+      await _requestPermissionLocationEvent(emit);
+    });
+    on<DefaultLocationServiceEvent>(
+        (LocationEvent event, Emitter<LocationState> emit) async {
+      await _defaultLocationServiceEvent(emit);
+    });
   }
 
   Future<void> _requestCurrentLocation(Emitter<LocationState> emit) async {
-    try {
-      final LocationDetails data = await _locationService.getCurrentLocation();
-      final String dateDisplay =
-          DateFormat('dd-MMM-yyyy').format(DateTime.now());
+    final bool isConnected = await InternetConnectionChecker().hasConnection;
+    //internet connected
+    if (isConnected) {
+      SharedPreferences prefs = await SharedPreferences.getInstance();
+      final bool isLoaded = prefs.getBool('isLoadedDialog') ?? false;
```

Comment:
When permission is denied, we can still show permission request dialog, therefore you do not need to create a Boolean to keep track if you have showed permission request dialog previously. Just use PermissionStatus to determine if we can show permission request dialog, or if we need to use default location.